front/components/assistant/conversation/input_bar/editor/extensions/MentionExtension.tsx

```diff
@@ -75,61 +87,118 @@ export const MentionExtension = Mention.extend<MentionExtensionOptions>({
-              new RegExp(
-                `((^@|\\s@)${escapeRegExp(suggestion.label)})(\\s|$)`,
-                "g"
-              )
-            ),
-          ].map((match) => {
-            return {
-              index: match.index,
-              text: match[1],
-              replaceWith: suggestion.label,
-              data: {
-                type: suggestion.type,
-                id: suggestion.id,
-                label: suggestion.label,
-                description: suggestion.description,
-                pictureUrl: suggestion.pictureUrl,
-              },
-            };
-          });
-        });
-        return results;
-      },
-      type: this.type,
-      getAttributes: (match: {
-        data: {
-          type: "agent" | "user";
-          label: string;
-          id: string;
-          description: string;
-          pictureUrl: string;
+  // define a custom Markdown tokenizer to recognize :mention: syntax
+  markdownTokenizer: {
+    name: "mention",
+    level: "inline", // inline element
+    // fast hint for the lexer to find candidate positions
+    start: (src) => src.indexOf(":mention"),
+    tokenize: (src) => {
+      const matchAgent = AGENT_MENTION_REGEX.exec(src);
+      const matchUser = USER_MENTION_REGEX.exec(src);
+      if (!matchAgent && !matchUser) {
+        return undefined;
+      }
+
+      if (matchAgent) {
+        return {
+          type: "mention", // token type (must match name)
+          raw: matchAgent[0], // full matched string
+          mentionType: "agent",
+          attrs: {
+            id: matchAgent[2],
+            label: matchAgent[1],
+          },
         };
-      }) => {
+      }
+
+      if (matchUser) {
         return {
-          type: match.data.type,
-          label: match.data.label,
-          id: match.data.id,
-          description: match.data.description,
-          pictureUrl: match.data.pictureUrl,
+          type: "mention", // token type (must match name)
+          raw: matchUser[0], // full matched string
+          mentionType: "user",
+          attrs: {
+            id: matchUser[2],
+            label: matchUser[1],
+          },
         };
+      }
+
+      return undefined;
+    },
+  },
+
+  // Parse Markdown token to Tiptap JSON
+  parseMarkdown: (token) => {
+    return {
+      type: "mention",
+      attrs: {
+        type: token.mentionType,
+        id: token.attrs.id,
+        label: token.attrs.label,
       },
-    });
+    };
+  },
 
-    return [pasteRule];
+  addProseMirrorPlugins() {
+    const { owner } = this.options;
+    const editor = this.editor;
+
+    return [
+      new Plugin({
+        key: mentionPastePluginKey,
+        props: {
+          handlePaste: (view, event, slice) => {
+            // Get text from the slice after TipTap processing.
+            const text = slice.content.textBetween(0, slice.content.size, "\n");
+
+            // Only process if text contains @.
+            if (!text.includes("@")) {
```

Comment:
this will trigger when you copy paste an email address, would it be too much if we try to avoid triggering for this case?